app/api/items/fetch_active_attempt.go

```diff
@@ -13,6 +13,60 @@ import (
 	"github.com/France-ioi/AlgoreaBackend/app/token"
 )
 
+// swagger:operation PUT /items/{item_id}/active-attempt items itemActiveAttemptFetch
+// ---
+// summary: Fetch an active attempt
```

Comment:
> It does actually generate (always, it is not a get-or-insert) a task-token for a given

Thinking this way, we can say that even GET requests (without any side effects) do always generate a new JSON response... For me, task tokens are just signed responses.

Reply:
Generating something is indeed just a "GET" but side effect is more on the actual data/db side. 
Here, do we change things except the side creations in case things were not created yet? From what I see, not really. So it could be a GET. So do you agree we should change this service to a GET?

Reply:
It also updates sStartDate and sLastActivity in groups_attemts & user_items on each request. So I'm not sure.

Reply:
I change the route and the service name. How do you think, is it better now?

Reply:
Actually, even it 90% of the code is dedicated to the specific case of not existing users_items and/or groups_attempts, this is just a minor case among the usage of this service. This service is just used to get the token.   Maybe you are biased by the fact that most of the code is for this case where `activeAttemptID` is nil (which should maybe moved as an independent function anyway).

---

For the URL, ` PUT /items/{item_id}/active-attempt` is still pretty wrong as a name for this service, from the user (of the service) point of view, which is used to get a "task token".  
If I'm a dev looking for the service to get a task-token, reading the service list, I'll never find that it is this one. 
The sStartDate is even not changed when it is not the first time. The `sLastActivity` is more a side effect of the GET. 

My best proposals for now are:
- `GET /items/:item_id/task-token`: we consider we ask for a task token
- `POST /items/:item_id/task-token`: we consider we create a task token given the input data

After checking as some public APIs with tokens, the GET has my preference.

Reply:
done